src/channel.js

```diff
@@ -723,6 +723,22 @@ export class Channel {
 	}
 
 	_initializeState(state) {
+		// add the Users
+		console.log('state', state);
+		if (state.members) {
+			for (const m of state.members) {
+				console.log('m', m);
+				this._client.state.updateUserReference(m.user, this.cid);
```

Comment:
poor man encapsulation here, please use `this.getClient()` instead of `this._client` (the getter is used to do sanity checks eg. check that the channel is not dandling after a client.disconnect)

Reply:
done, but the disconnect flow for channels seems broken